src/rwe/Unit.cpp

```diff
@@ -128,6 +142,7 @@ namespace rwe
         if (hitPoints + deltaHp >= maxHitPoints)
         {
             hitPoints = maxHitPoints;
+            lifeState = LifeState::Alive;
```

Comment:
In TA a unit can reach maximum health before finishing being built due to health regeneration (not yet implemented in RWE). This change would mean that a unit would finish being built when reaching maximum health instead of when all the required build effort has been put in.